### PR TITLE
RHCLOUD-35424: Add widget registry to the frontend resource

### PR DESCRIPTION
**Pull request by florkbr**

### CODE REVIEW

File: api/v1alpha1/frontend_types.go [hunk 1]
```diff
@@ -58,6 +58,24 @@ type ServiceTile struct {
...
+}
+
+type WidgetConfig struct {
+	Icon        string           `json:"icon" yaml:"icon"`
+	Title       string           `json:"title" yaml:"title"`
+	Permissions []Permission     `json:"permissions,omitempty" yaml:"permissions,omitempty"`
+	HeaderLink  WidgetHeaderLink `json:"headerLink,omitempty" yaml:"headerLink,omitempty"`
+}
+
+type WidgetEntry struct {
```
Comment (florkbr) [inline api/v1alpha1/frontend_types.go:73]: @Hyperkid123 do you think we need to include the `Default`s for the widgets here too? See https://github.com/RedHatInsights/chrome-service-backend/blob/main/rest/service/dashboardTemplate.go#L29

Comment (Hyperkid123) [reply api/v1alpha1/frontend_types.go:73]: I think so and for each layout variant. Good catch.

Comment (florkbr) [reply api/v1alpha1/frontend_types.go:87]: Agree. For now added the defaults to the struct

**Review by Hyperkid123 — APPROVED** [review]

**Comment by adamrdrew** [comment]
LGTM

**Pull request merged**